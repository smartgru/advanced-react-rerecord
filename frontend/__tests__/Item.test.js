import { render } from '@testing-library/react';
import ItemComponent from '../components/Item';

const fakeItem = {
  id: 'ABC123',
  title: 'A Cool Item',
  price: 4000,
  description: 'This item is really cool!',
  image: 'dog.jpg',
  largeImage: 'largedog.jpg',
};

describe('<Item/>', () => {
  fit('renders and matches the snapshot', () => {
    const { container } = render(<ItemComponent item={fakeItem} />);
    expect(container).toMatchSnapshot();
  });

  it('renders the image properly', () => {
    const { container } = shallow(<ItemComponent item={fakeItem} />);
    const img = container.find('img');
    expect(img.props().src).toBe(fakeItem.image);
    expect(img.props().alt).toBe(fakeItem.title);
  });

  it('renders the pricetag and title', () => {
    const { container } = shallow(<ItemComponent item={fakeItem} />);
    const PriceTag = container.find('PriceTag');
    expect(PriceTag.children().text()).toBe('$40');
    expect(container.find('Title a').text()).toBe(fakeItem.title);
  });

  it('renders out the buttons properly', () => {
    const { container } = shallow(<ItemComponent item={fakeItem} />);
    const buttonList = container.find('.buttonList');
    expect(buttonList.children()).toHaveLength(3);
    expect(buttonList.find('Link')).toHaveLength(1);
    expect(buttonList.find('AddToCart').exists()).toBe(true);
    expect(buttonList.find('DeleteItem').exists()).toBe(true);
  });
});
