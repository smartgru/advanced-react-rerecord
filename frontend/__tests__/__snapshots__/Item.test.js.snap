// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Item/> renders and matches the snapshot 1`] = `
<div>
  <div
    class="ItemStyles__Item-sc-16pk14u-0 dXyHNM"
  >
    <img
      alt="dogs are best"
      src="dog.jpg"
    />
    <h3
      class="Title-sc-16nq74k-0 cBlaqK"
    >
      <a
        href="/item?id=abc123"
      >
        dogs are best
      </a>
    </h3>
    <span
      class="PriceTag-nwbk6t-0 hBhTMo"
    >
      $50
    </span>
    <p>
      dogs
    </p>
    <div
      class="buttonList"
    >
      <a
        href="update?id=abc123"
      >
        Edit ✏️
      </a>
      <button
        type="button"
      >
        Add
         To Cart 🛒
      </button>
      <button
        type="button"
      >
        Delete This Item
      </button>
    </div>
  </div>
</div>
`;
